### PR TITLE
Create Bag with initial elements

### PR DESCRIPTION
**Pull request by ra1028**
Hi✋ 
I made it possible to create Bag with initial elements, and CompositeDisposable also improved accordingly.
Probably conflicts with #608, but I fix after either is merged.

#### Checklist
- [x] Updated CHANGELOG.md.

### CODE REVIEW

File: Sources/Bag.swift [hunk 1]
```diff
@@ -14,12 +14,22 @@ public struct Bag<Element> {
...
+	public init() {
+		elements = ContiguousArray()
+		tokens = ContiguousArray()
+		nextToken = Token(value: 0)
+	}
+
+	public init<S: Sequence>(_ elements: S) where S.Iterator.Element == Element {
+		nextToken = Token(value: UInt64(elements.underestimatedCount))
+		tokens = ContiguousArray(0..<nextToken.value)
+		self.elements = ContiguousArray(elements)
```
Comment (NachoSoto) [inline Sources/Bag.swift]: Be consistent here: either self for all or none.

File: Sources/Disposable.swift [hunk 2]
```diff
@@ -140,12 +140,7 @@ public final class CompositeDisposable: Disposable {
 	public init<S: Sequence>(_ disposables: S)
 		where S.Iterator.Element == Disposable
 	{
-		var bag: Bag<Disposable> = Bag()
-
-		for disposable in disposables {
-			bag.insert(disposable)
-		}
-
+		let bag = Bag(disposables)
```
Comment (NachoSoto) [inline Sources/Disposable.swift]: 👍

File: Sources/Bag.swift (hunk 1 above)
Comment (ra1028) [reply Sources/Bag.swift]: will do.
`elements` is used as a parameter name, so I'll add` self` to all.

File: Sources/Bag.swift [hunk 3]
```diff
@@ -14,12 +14,22 @@ public struct Bag<Element> {
...
 
-	public init() {}
+	public init() {
+		elements = ContiguousArray()
+		tokens = ContiguousArray()
+		nextToken = Token(value: 0)
+	}
+
+	public init<S: Sequence>(_ elements: S) where S.Iterator.Element == Element {
+		self.nextToken = Token(value: UInt64(elements.underestimatedCount))
```
Comment (andersio) [inline Sources/Bag.swift]: `underestimatedCount` is not guaranteed to equal the exact number of elements. We should derive this from the `ContiguousArray` constructed from `elements`.

Comment (ra1028) [reply Sources/Bag.swift]: Ah, true.

Comment (ra1028) [reply Sources/Bag.swift]: Amended:+1:

**Comment by mdiep** [comment]
Thanks for the PR!

**Pull request merged**